### DISCUSSION TITLE
Need help using LuaRocks on Windows, to build Lua DLLs for 64-bit Windows, Lua 5.3

### DISCUSSION THREAD

**Discussion by simonx222** [keplerproject/luarocks#1547] [Q&A]
I'm in the process of converting a Windows desktop application from 32-bit to 64-bit.  The application supports Lua as an embedded scripting language, using version 5.3.   I need new 64-bit, Lua 5.3 versions of all of the Lua DLLs.  I have already found the code modules I need for CD, IM, IUP, LFS and LUACOM (thank you Antonio Scuri!).  However, there are numerous DLLs I’m still missing these:

•         MD5 dlls (core.dll and des56.dll)
•         MIME dll (core.dll)
•         ODBC dll (core.dll)
•         Windows Sockets dll (core.dll)
•         SQLite3 dll (lsqlite3.dll)

First - can anyone point me at binary versions of these, for x64 Windows (Lua 5.3 compatible)?  If not, I presume I need to build them using LuaRocks? 

Unfortunately, I have tried using LuaRocks to build them and can't get it to work.  The all-in-one package seems to be designed for x86, Lua 5.1 - and in any case I got errors with it when I tried to build.  So I tried using luarocks-3.9.2-windows-64.zip instead.  This produced 2 files: LuaRocks.exe and LuaRocks-admin.exe.  I had a 64-bit build of Lua 5.3.  I opened the Visual Studio 2022 x64 Native Tools command prompt.  I put Lua53.dll, luarocks.exe and luarocks-admin.exe on the PATH.  I also created a rocktree structure corresponding to the structure outlined in "Rocks Repositories" at https://github.com/luarocks/luarocks/wiki/Rocks-repositories.  And I tried putting folders from this on LUA_PATH and LUA_CPATH.  But it's not working.  First, the luarocks.exe seems to only want to use Lua 5.4.  There are configuration options to set the Lua version, but I can't get any of them to work.  Nothing seems to have any persistent effect.  I also can't work out how to change values that seem to refer to Lua 5.4.  For example, if I type "luarocks config", it includes this line:

LUALIB="lua5.4.lib",
 
How do I change it to "Lua53.lib" which is presumably what it should be?

If I try running "luarocks install md5" (for example), I get the message "Warning: Lua 5.4 interpreter not found at ... "<my path>.  I also get the message "Error: Lua header found at D:\My\Lua64\Lua64\unzip\include does not match Lua version 5.4. You may want to override this by configuring LUA_INCDIR."  I have tried configuring LUA_INCDIR but it doesn't seem to make any difference.

All help very much appreciated.

**Comment by hishamhm** [comment]
> How do I change it to "Lua53.lib" which is presumably what it should be?

To change your default Lua version, try `luarocks config lua_version 5.3`

**Comment by simonx222** [comment]
Thanks hishamhm.  Pretty sure I tried that.  However, since I posted I have given up trying to use luarocks-3.9.2-windows-64.zip and instead have been using luarocks-3.6.0-windows-64.zip, which seemingly defaults to 5.3, and I've been having more luck with that.  It looks like I can use this version to build the DLLs I need.
